report/report.py

```diff
@@ -0,0 +1,213 @@
+from sisyphus import *
+import sys
+import os
+import getpass
+import pprint
+import gzip
+from typing import Dict, Union, Callable
+import time
+import json
+
+_Report_Type = Dict[str, Union[tk.Path, any]]
+
+
+class ReportResultsJob(Job):
+    """
+    Job to report results either via mail and/or just condensed in a file
+    """
+
+    def __init__(
+        self,
+        name: str,
+        report: _Report_Type,
+        report_dir: tk.Path = None,
+        report_format: Callable[[_Report_Type], str] = None,
+        recipe_path=Path("."),
+    ):
+        """
+
+        :param name:
+        :param dict report: dictionary containing the report files
+        :param report_dir:
+        :param (report: dict) -> List[str] report_format:
+        :param recipe_path:
+        """
+        self.name = name
+        self.report = report
+        self.report_dir = report_dir
+        self.report_format = report_format
+        self.recipe_path = recipe_path
+
+        self.out_report = self.output_path("report.gz")
```

Comment:
idea: maybe the job could have a flag to either compress the report file or not..